network/openssl_cert.py

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Yanis Guenane <yanis+ansible@guenane.org>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.basic import *
+from OpenSSL import crypto
+from random import randint
+
+import os
+
+
+DOCUMENTATION = '''
+---
+module: openssl_cert
+author: "Yanis Guenane (@Spredzy)"
+version_added: "2.2"
+short_description: Generate OpenSSL Certificates
+description:
+    - "This module allows one to (re)generates OpenSSL certificates. It implements a notion
+       of provider (ie. 'self-signed', 'letsencrypt'), it can be further extended.
+       It uses the pyOpenSSL python library to interact with openssl."
+requirements:
+    - "python-pyOpenSSL"
+options:
+    state:
+        required: false
+        default: "present"
+        choices: [ present, absent ]
+        description:
+            - Whether the certificate should exist or not, , taking action if the state is different from what is stated.
+    name:
+        required: true
+        description:
+            - Name of the generated OpenSSL certificate
+    provider:
+        required: true
+        choices: ['self-signed', 'letsencrypt' ]
+        description:
+            -  Name of the provider to use to generate/retrieve the OpenSSL certificate
+    force:
+        required: false
+        default: False
+        choices: [ True, False ]
+        description:
+            - Should the certificate be forced regnerated by this ansible module
+    path:
+        required: true
+        description:
+            - Name of the folder in which the generated OpenSSL certificate  will be written
+    csr:
+        required: true
+        description:
+            - Path to the Certificate Signing Request (CSR) needed to generate this certificate
+
```

Comment:
can you add options for setting the owner, group, and mode for the generated certificate?
eg,
````
  owner:
    description:
      - The owner id of the generated certificate.
    required: false
    default: null
  group:
    description:
      - The group id of the generated certificate.
    required: false
    default: null
  mode:
    description:
      - The permissions of the generated certificate file.
    required: false
    default: null
````